### PR TITLE
add agent builder example

### PR DESCRIPTION
**Pull request by Osier-Yi**
## Description

add agent builder example

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: examples/agent_builder/auto-discussion.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+discussion_agent = DialogAgent(
+    name="agent_builder",
+    sys_prompt="You're a helpful assistant.",
+    model="my_post_api",
+    # model="gpt-3.5-turbo",  # replace by your model config name
+)
+
+
+max_round = 1
```
Comment (DavdGao) [inline examples/agent_builder/auto-discussion.py:45]: Maybe larger than 1 round?

File: examples/agent_builder/auto_diss_prompt.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        participant_type, characteristic = match.groups()
+        participants_dict[
+            participant_type.strip().replace(" ", "_")
+        ] = characteristic.strip()
+    result["Participants"] = participants_dict
+
+    return result
+
+
+if __name__ == "__main__":
```
Comment (DavdGao) [inline examples/agent_builder/auto_diss_prompt.py:84]: It's suggested to rename this file into "utils.py" and remove the main entry here.

**Review by DavdGao — COMMENTED** [review]
Please see inline comments, and reformat the code according to pre-commit.

File: examples/agent_builder/auto-discussion.py (hunk 1 above)
Comment (Osier-Yi) [reply examples/agent_builder/auto-discussion.py]: fix; change to 2

File: examples/agent_builder/auto_diss_prompt.py (hunk 2 above)
Comment (Osier-Yi) [reply examples/agent_builder/auto_diss_prompt.py]: fix; change the name to utils.py, and delete the main entry

File: examples/agent_builder/auto-discussion.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+"""A simple example for auto discussion: the agent builder automatically\
+ set up the agents participating the discussion ."""
+from utils import load_txt, extract_scenario_and_participants
+import agentscope
+from agentscope.agents import DialogAgent
+from agentscope.pipelines.functional import sequentialpipeline
+
+model_configs = [
+    {
+        "type": "openai",
```
Comment (DavdGao) [inline examples/agent_builder/auto-discussion.py:11]: Please update keys "type", "name" into "config_name", "model_type" and "model_name" as [README.md](https://github.com/modelscope/agentscope/blob/main/README.md#openai-api-config)

File: examples/agent_builder/auto-discussion.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        "name": "gpt-3.5-turbo",
+        "api_key": "xxx",  # Load from env if not provided
+        "organization": "xxx",  # Load from env if not provided
+        "generate_args": {
+            "temperature": 0.5,
+        },
+    },
+    {
+        "type": "post_api",
```
Comment (DavdGao) [inline examples/agent_builder/auto-discussion.py:20]: Same as above

File: examples/agent_builder/auto-discussion.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    },
+]
+agentscope.init(model_configs=model_configs)
+
+
+# init agent_builder
+agent_builder = DialogAgent(
+    name="agent_builder",
+    sys_prompt="You're a helpful assistant.",
+    model="my_post_api",
```
Comment (DavdGao) [inline examples/agent_builder/auto-discussion.py:33]: into "model_config_name"

File: examples/agent_builder/auto-discussion.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    # model="gpt-3.5-turbo",  # replace by your model config name
+)
+
+discussion_agent = DialogAgent(
+    name="agent_builder",
+    sys_prompt="You're a helpful assistant.",
+    model="my_post_api",
```
Comment (DavdGao) [inline examples/agent_builder/auto-discussion.py:40]: Same as above

File: examples/agent_builder/auto-discussion.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+)
+settings = agent_builder(x)
+scenario_participants = extract_scenario_and_participants(settings["content"])
+
+# set the agents that participant the discussion
+agents = [
+    DialogAgent(
+        name=key,
+        sys_prompt=val,
+        model="my_post_api",
```
Comment (DavdGao) [inline examples/agent_builder/auto-discussion.py:62]: same as above

**Review by DavdGao — COMMENTED** [review]
Please see inline comments, and test example with the latest agentscope package.

File: examples/agent_builder/auto-discussion.py (hunk 4 above)
Comment (Osier-Yi) [reply examples/agent_builder/auto-discussion.py]: fix

File: examples/agent_builder/auto-discussion.py (hunk 7 above)
Comment (Osier-Yi) [reply examples/agent_builder/auto-discussion.py]: fix

**Review by DavdGao — APPROVED** [review]
LGTM. Thanks to your contribution!

**Pull request merged**